### ISSUE TITLE
Unexpected EOF causes panic

### ISSUE THREAD

**Issue by rcgenova** [hashicorp/terraform#8761]
Received the following error due to a nested list issue running a plan against 0.7.0.

```
Error running plan: 8 error(s) occurred:

* element: element() may only be used with flat lists, this list contains elements of type list in:
${element(var.private_subnet_names, count.index)}
```
[0.7.0 run in Atlas](https://atlas.hashicorp.com/terraform/rcgenova/environments/google-us-central1-prod/changes/runs/26)

Updating the Terraform version to 0.7.1 in Atlas led to the following errors and a panic:

```
Error refreshing state: 8 error(s) occurred:

* google_compute_firewall.allow-ssh: unexpected EOF
* google_compute_firewall.allow-internal: unexpected EOF
* google_compute_subnetwork.public.1: unexpected EOF
* google_compute_subnetwork.public.0: unexpected EOF
* google_compute_subnetwork.public.2: unexpected EOF
* google_compute_subnetwork.private.1: unexpected EOF
* google_compute_subnetwork.private.2: unexpected EOF
* google_compute_subnetwork.private.0: unexpected EOF
panic: interface conversion: interface is nil, not string
```

[Full plan output](https://gist.github.com/rcgenova/7136290d7c68aac04a4820961ec6de06)

[0.7.1 run in Atlas](https://atlas.hashicorp.com/terraform/rcgenova/environments/google-us-central1-prod/changes/runs/27) (for crash.log)

The full configuration can be found [here](https://github.com/hashicorp/best-practices/tree/f-google-provider), but the nested list issues have since been fixed.

I have yet to succeed in recreating this with a simplified, deterministic set of steps and configuration.

**Comment by kafis** [comment]
I have similiar problems also with the following resources
 

- google_compute_project_metadata
- google_compute_address
- google_compute_network

after upgrading from 0.6.2 to 0.7.5

**Comment by kafis** [comment]
I made a comparison of the state files for the "google_compute_firewall" resource, that I created in 0.6 and 0.7

The differences are, that the state file for 0.7 contains attributes, that are not included in the state file for 0.6

To be explicit:
My resource does not contain source or target tags. So in the state file of 0.6 they are not present.
In the state file of 0.7 they are present, but set to the value "0"

Also some more attributes appeared, but I dont suspect them:

 - meta
 - tainted

Might it be, that the new version of terraform expects these new attributes to be present in the state file, altough they were not set previously by version 0.6, because they are not used in my resource definition anyway?
